### ISSUE TITLE
Exported Keras Model classifying incorrectly 

### ISSUE THREAD

**Issue by Tailorware** [googlecreativelab/teachablemachine-community#57]
The downloaded Keras model classifies images differently than the Online Version. Probably the Image preprocessing in the provided Python Snippet is different from the one used on the website.

I build a two-class classifier for 3-channel vibration data (represented as RGB images) that works perfectly in the online version of Teachable Machine. Impressive Work by the way!
But if I export the model to Keras and try the same predictions in Google Colab it always predicts the same class.
Link to Teachable Machine:
https://teachablemachine.withgoogle.com/models/dhRmTcji/

Link to Colab:
https://colab.research.google.com/drive/12LGL5LDGNdoESE7L7tyWYW-YakzNOOhF

The Colab Notebook also contains links to download training and test set.

This is most likely a preprocessing issue.

It would be great if you could have a look. 
Thank you in advance!

**Comment by Tailorware** [comment]
After reading [https://github.com/googlecreativelab/teachablemachine-community/issues/3](url) I tried to normalize between -1 and 1, which improves the classification accuracy but it does not fix the problem entirely. The classfier is still wrong on half of the samples of the "bad" class.

Changed line of code:
 normalized_image_array = (image_array.astype(np.float32) / 127.0) -1

**Comment by GautamBose** [comment]
Hi! I went through your Colab, and noticed the same classification performance issue. 

However, I trained a new model in teachable machine with images in the training folder you provided, re-exported to keras, and tried the test samples again with the corrected normalization between -1 and 1. The performance was accurate. I suspect the keras model loaded into the Colab isn't the most up to date with the one trained in TM2. 

I've modified the drive link in a copy of the Colab with my newly exported keras model here: 

[](https://drive.google.com/file/d/1rJ5H7xTsXimwXD_D1_i5uPwdL9M4gkxJ/view?usp=sharing)

My TM model is here:

[](https://teachablemachine.withgoogle.com/models/WwtKDbVK/
)

The normalization is updated in the snippet: #58
